modules/navigation/nav_utils.h

```diff
@@ -150,6 +150,34 @@ struct ClosestPointQueryResult {
 	RID owner;
 };
 
+enum PathfindingAlgorithm {
+	PATHFINDING_ALGORITHM_ASTAR = 0,
+};
+
+struct PathQueryParameters {
+	PathfindingAlgorithm pathfinding_algorithm = PATHFINDING_ALGORITHM_ASTAR;
```

Comment:
Any need to put this in in this PR if there is currently only one pathfind method?

Reply:
As discussed on rocket chat (better to keep on the PR for history purposes) : 

> About the enum / pathfinding algorithms included in the pr. I included them because I think if multiple ppl start working on different algo we may end up with x-implementations all over the place. This way everyone knows the access point to add the new stuff. E.g. if Acheron and I start working on two different things we would just add our algos to the existing enum. This was the general idea why this was added while we have no different pathfinding implemented.

That seems fair, and is up to you, not something that is a show stopper. :slightly_smiling_face: 

I just try where possible to refer to the guidelines, and being wary of the section on futureproofing:
https://docs.godotengine.org/en/stable/community/contributing/best_practices_for_engine_contributors.html

Personally I also tend to err on the side of future proofing, but try and hold back a bit for Godot because of the different philosophy. :slightly_smiling_face:

Reply:
Enum and this was included so if multiple ppl start work on new algorithms they would know where to add them instead of ending up with x-different implementations all over the place. I know DarkKilauea is just waiting for this pr to be merged to start adding one better algo for the navigation jumplinks and I also want to start work, might be others too.